src/components/SearchInput/SearchInput.jsx

```diff
@@ -1,21 +1,22 @@
-import React, { useState, useEffect } from 'react';
-import { DebounceInput } from 'react-debounce-input';
+import React, { useState, useEffect, useCallback } from 'react';
+import { useDebounce } from '../../hooks/useDebounce.jsx';
+import { dataService } from '../../services/dataService';
 
 import styles from './SearchInput.module.scss';
 import SearchIcon from '../../assets/images/search-icon.png';
 import SearchLocationIcon from '../../assets/images/search-location-icon.png';
-import { NUMBER_OF_RESULTS } from '../../constants/forecaApi';
-import { dataService } from '../../services/dataService';
 
 function SearchInput() {
   const [searchQuery, setSearchQuery] = useState('');
   const [searchResults, setSearchResults] = useState([]);
 
+  const debouncedSearchQuery = useCallback(useDebounce(searchQuery, 1000));
```

Comment:
useCallback accepts a function... and a list of dependencies, if you don't want it to run every time returning you new value. In your case you are passing a string to it from what I see. So I'm a bit confused.

Reply:
Yes. I rewrote function and forget remove this one. Fixed.